### ISSUE TITLE
no-unused-variable: Not detecting unused in object deconstruction

### ISSUE THREAD

**Issue by OliverJAsh** [palantir/tslint#1606]
### Bug Report
- **TSLint version**: `3.15.1`
- **TypeScript version**: `2.0.2`
- **Running TSLint via**: Visual Studio Code
#### TypeScript code being linted

``` ts
const foo = () => ({ y: 1 });

(() => {
    const { y } = foo();
})()

(() => {
    const { y } = foo();
})()
```

with `tslint.json` configuration:

``` json
{
    "rules": {
        "no-unused-variable": true
    }
}
```
#### Actual behavior

No error
#### Expected behavior

`y` is unused in both IIFEs.

**Comment by adidahiya** [comment]
we're deprecating this rule in v4.0, so I'm marking this as "won't fix". check out https://github.com/palantir/tslint/issues/1481 for reasoning & alternatives.

**Comment by andy-ms** [comment]
Should be fixed by #2235.

**Comment by ajafff** [comment]
This was fixed in the v5.0.0 release
